angularjs/angular-resource.d.ts

```diff
@@ -71,12 +71,12 @@ declare module angular.resource {
     // Also, static calls always return the IResource (or IResourceArray) retrieved
     // https://github.com/angular/angular.js/blob/v1.2.0/src/ngResource/resource.js#L538-L549
     interface IResourceClass<T> {
-        new(dataOrParams? : any) : T;
-        get(): T;
-        get(params: Object): T;
-        get(success: Function, error?: Function): T;
-        get(params: Object, success: Function, error?: Function): T;
-        get(params: Object, data: Object, success?: Function, error?: Function): T;
+        new(dataOrParams? : any) : IResource<T>;
+        get(): IResource<T>;
```

Comment:
um? is this right?
now typescript is support intersection types. https://github.com/Microsoft/TypeScript/pull/3622
I think we should use `get(): T & IResource<T>;`. isn't it?